core/trino-main/src/main/java/io/trino/operator/scalar/MathFunctions.java

```diff
@@ -1176,6 +1176,15 @@ public static boolean isNaN(@SqlType(StandardTypes.DOUBLE) double num)
         return Double.isNaN(num);
     }
 
+    @Description("Test if value is not-a-number")
+    @ScalarFunction("is_nan")
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean isNaNReal(@SqlType(StandardTypes.REAL) long value)
+    {
+        float floatValue = intBitsToFloat((int) value);
```

Comment:
i'd use `toIntExact` here.

Reply:
I thought about it, but seems like this is how `isNaN` usually tested (at this class, at `io.trino.type.RealOperators` and at `io.trino.spi.type.RealType`). I can change it in all places (on a separated PR), WDYT?

Reply:
Yes, we're not consistent. Sometimes we do unchecked cast (`(int)`) and sometimes we do a checked one (`toIntExact`).  In new code, please use `toIntExact` unless there is a compelling reason not to.

Reply:
done